music21/romanText/writeRoman.py

```diff
@@ -0,0 +1,552 @@
+    'm0 b1 a: viio64'
+
+    OMIT_FROM_DOCS
+
+    Users can do these insertions themselves, but don't need to:
+
+    >>> m = stream.Measure()
+    >>> m.insert(0, rn)
+    >>> rnWriterFromMeasure = romanText.writeRoman.RnWriter(m)
+    >>> rnWriterFromMeasure.combinedList == rnWriterFromRn.combinedList
+    True
+
+    >>> p = stream.Part()
+    >>> p.insert(0, m)
+    >>> rnWriterFromPart = romanText.writeRoman.RnWriter(p)
+    >>> rnWriterFromPart.combinedList == rnWriterFromMeasure.combinedList
+    True
+
+    >>> s = stream.Score()
+    >>> s.insert(0, m)
+    >>> rnWriterFromScoreWithoutPart = romanText.writeRoman.RnWriter(p)
+    >>> rnWriterFromScoreWithoutPart.combinedList == rnWriterFromMeasure.combinedList
+    True
+    '''
+
+    def __init__(self,
+                 obj: base.Music21Object,
+                 ):
+
+        self.composer = 'Composer unknown'
+        self.title = 'Title unknown'
+
+        if obj.isStream:
+
+            if isinstance(obj, stream.Opus):
+                constituentElements = [RnWriter(x) for x in obj]
+                self.combinedList = []
+                for scoreOrSim in constituentElements:
+                    for x in scoreOrSim.combinedList:
+                        self.combinedList.append(x)
+                    self.combinedList.append('\n')  # one between scores
+                return
+
+            elif isinstance(obj, stream.Score):
+                if obj.parts:
+                    self.container = obj.parts[0]
+                else:  # score with no parts
+                    self.container = obj
+
+            elif isinstance(obj, stream.Part):
+                self.container = obj
+
+            elif isinstance(obj, stream.Measure):
+                self.container = stream.Part()
+                self.container.insert(0, obj)
+
+            else:  # A stream, but not a measure, part, or score
+                self._makeContainer(obj)
+
+            if obj.metadata:  # Check the obj (not container) for metadata if obj is a stream
+                self.prepTitle(obj.metadata)
+                if obj.metadata.composer:
+                    self.composer = obj.metadata.composer
+
+        else:  # Not a stream
+            self._makeContainer([obj])
+
+        self.combinedList = [f'Composer: {self.composer}',
+                             f'Title: {self.title}',
+                             'Analyst: ',
+                             'Proofreader: ',
+                             '']  # One blank line between metadata and analysis
+        # Note: blank analyst and proof reader entries until supported within music21 metadata
+
+        if not self.container.getElementsByClass('TimeSignature'):
+            self.container.insert(0, meter.TimeSignature('4/4'))  # Placeholder
+
+        self.currentKeyString: str = ''
+        self.prepSequentialListOfLines()
+
+    def _makeContainer(self,
+                       obj: Union[stream.Stream, list]):
```

Comment:
For Python pre-3.9 (i.e., music21 currently), list needs to be List and imported from typing module.  typing.List is different from the builtin list until Py3.9